source/Concepts/Basic/About-Launch.rst

```diff
@@ -7,30 +7,48 @@ Launch
 
-The purpose of the ``launch`` system in ROS 2 is to provide the means to orchestrate the execution of such a system. This tool uses _launch files_ to describe the system and its configuration, but not limited to which programs to run, how to run them and which arguments to use for them. These descriptions are then executed, starting processes, monitoring them, reporting on their state, and even reacting to their behavior.
+``launch`` provides the means to orchestrate the execution of ROS 2 systems, in ROS 2 terms.
+_Launch files_ describe the orchestration procedure, including but not limited to which nodes to run, how to run them, and which arguments to use for them.
+These descriptions are then executed, starting processes, monitoring them, reporting on their state, and even reacting to their behavior.
 
-System descriptions
--------------------
+Moving parts
+------------
 
-``launch`` works with _descriptions_, which, for the most part, amount to sequences of _actions_ executing in a _context_ or scope. All actions have side effects, which may be limited to the context in which they are running or extend to the larger execution environment (e.g. the local operating system). These side effects will typically outlive the context in which the corresponding action was run, which may be obvious for an action that changes an environment variable but not for an action that executes a process. Actions may also generate _events_, signals that are propagated through the context, that may be handled by _event handlers_.
+``launch`` works with _actions_, abstract representations of computational procedures with side effects on its execution environment. Logging to standard output, executing a program, shutting down ``launch`` itself, are examples of actions. An action may also encompass another action or a collection thereof to describe more complex functionality. A collection of actions makes up the _description_ of a procedure that ``launch`` can take and execute. These descriptions are typically read from source files, so called _launch files_, which may be written in Python, or using specific XML or YAML syntax.
 
-Actions alone allow for static descriptions that act upon and react to the environment in predefined ways. For a description to adapt to external input, _conditions_ and _substitutions_ exist. The former are boolean predicates, used for conditional actions, while the latter resemble text macros, useful for dynamic action configuration.
+While the extent of the execution environment of an action depends on its nature e.g. logging to standard output is circumscribed to the ``launch`` process whereas executing a program reaches out to the host operating system, ``launch`` maintains an execution _context_ in which these actions take place and through which these can interact between them and with the user.
```

Comment:
I think I could use an example of context after this before we start introducing configuration variables and events.

Reply:
Context is just the logical space in which actions, and ``launch`` entities in general, live. There are no various contexts. That said, I guess we could have an example where we explain where are the boundaries for nested scopes, but way down below.

Reply:
I expanded this paragraph in https://github.com/mhidalgo-bdai/ros2_documentation/pull/1/commits/1ecbbac0e4a1339cc85886fb9a60f50c9ede4214. Hopefully it's clearer now.